util/cmd/cmd.go

```diff
@@ -87,6 +88,12 @@ func ParseLabels(labelSpec interface{}) (map[string]string, error) {
 		if len(labelSpec[0]) == 0 {
 			return nil, fmt.Errorf("unexpected empty label key")
 		}
+
+		if _, reserved := common.ReservedWorkflowLabelKeys[labelSpec[0]]; reserved {
+			log.Warnf("Ignoring label with key '%s' as it is reserved", labelSpec[0])
+			continue
+		}
```

Comment:
Not super happy with this solution as it only checks CLI submitted labels. However, checking at the WF validation stage is problematic because legitimate reserved labels might be set by Argo sometime before the validation stage, requiring some refactor to correctly validate at that stage.

Reply:
Yes a CLI-only check would indeed be insufficient in many ways. A validation stage check or an informer change check would be most appropriate, but could also be worked around by a user with sufficient privileges. 

Therefore, I don't think this approach is doable or that the issue itself is even solveable in k8s. SAs and other identity mechanisms within k8s should be used for this, as those are properly supported authN in k8s.